### PR TITLE
Add scraper for rickbayless.com

### PR DESCRIPTION
**Pull request by joeygerovac**
## Add scraper for rickbayless.com

Adds support for scraping recipes from rickbayless.com.

### Site structure
Rick Bayless's site uses custom HTML markup with no JSON-LD structured 
data, requiring a dedicated scraper. Recipe data is stored in semantic 
class names:

- Title: `<h1>` inside `div.page-header`
- Description: `div.recipe-description`
- Ingredients: `<li itemprop="ingredients">` with separate spans for 
  quantity/unit, name, and preparation notes
- Instructions: `<p>` tags inside `div.recipe-instructions`

### Testing
Tested against multiple recipes including:
- Simple recipes (classic guacamole)
- Recipes with quantity-less ingredients ("Salt")
- Recipes with linked ingredient names (anchor tags)
- Single and multi-paragraph instruction sets
- Recipes with and without serving size data

All 1030 tests pass.

### Edge cases handled
- Ingredients with no quantity parse correctly
- Anchor tags within ingredient names are stripped to plain text
- `yields()` returns `None` when serving size div is empty
- `image()` falls back gracefully when no og:image tag exists

### CODE REVIEW

File: recipe_scrapers/rickbayless.py [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+        return "\n".join(
+            p.get_text().strip()
+            for p in instructions_div.find_all("p")
+            if p.get_text(strip=True)
+        )
+
+    def image(self):
+        og_image = self.soup.find("meta", {"property": "og:image"})
+        return og_image["content"] if og_image else None
+
```
Comment (jknndy) [inline recipe_scrapers/rickbayless.py]:
```suggestion
```
The built in `best_image` functionality (see [here](https://github.com/hhursev/recipe-scrapers/blob/50c8e20c133d2781290b729e9519b3d5d12591f9/recipe_scrapers/plugins/best_image.py)) returns the same result so we can exclude the custom code here.

File: recipe_scrapers/rickbayless.py [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+    def yields(self):
+        servings = self.soup.find("div", {"class": "recipe-servings"})
+        if not servings:
+            return None
+        text = servings.get_text(strip=True)
+        result = text.replace("Servings:", "").strip()
+        return result or None
```
Comment (jknndy) [inline recipe_scrapers/rickbayless.py]:
```suggestion
        return get_yields(result)

    def total_time(self):
        raise FieldNotProvidedByWebsiteException(return_value=None)

    def author(self):
        raise StaticValueException(return_value="Rick Bayless")
```

`yields`: There is a built in util that can be used to help standardize the output of the yields which helps maintain consistency in the output. There is a bug with this specific site that i'll be opening an issue for.
`total_time`: Since total time is a [mandatory field](https://docs.recipe-scrapers.com/contributing/in-depth-guide-scraper-functions/?h=mandatory#mandatory-functions) we should raise `FieldNotProvidedByWebsiteException` to alert that this field isn't available from this site
`author`: Another mandatory field here but this time we can raise `StaticValueException` with the author's name as it seems they are the sole contributor to the site.

File: recipe_scrapers/rickbayless.py [hunk 3]
```diff
@@ -0,0 +1,44 @@
+from ._abstract import AbstractScraper
+from ._exceptions import StaticValueException
```
Comment (jknndy) [inline recipe_scrapers/rickbayless.py]:
```suggestion
from ._exceptions import StaticValueException, FieldNotProvidedByWebsiteException
from ._utils import get_yields
```

File: tests/test_data/rickbayless.com/rickbayless.json [hunk 4]
```diff
@@ -0,0 +1,24 @@
+{
+  "canonical_url": "https://www.rickbayless.com/recipe/mango-guacamole/",
```
Comment (jknndy) [inline tests/test_data/rickbayless.com/rickbayless.json:3]:
```suggestion
  "author": "Rick Bayless",
  "canonical_url": "https://www.rickbayless.com/recipe/mango-guacamole/",
```

File: tests/test_data/rickbayless.com/rickbayless.json [hunk 5]
```diff
@@ -0,0 +1,24 @@
...
+    "About 2 tablespoons fresh lime juice",
+    "1 medium ripe mango, peeled, flesh cut from the pit and diced",
+    "Salt"
+  ],
+  "instructions_list": [
+    "Cut the avocados in half, running your knife around the pit from stem to blossom end and back up again. Twist the halves in opposite directions to free the pit, and pull the halves apart. Dislodge the pit, then scoop the avocado flesh into a large bowl. Coarsely mash the avocado with a large fork or potato masher. Rinse the onion under cold water, shake off the excess water, then add it to the avocado along with the serrano, cilantro and lime juice.",
+    "Mix in 2/3 of the diced mango. Taste and season with salt. If not using immediately, cover with plastic wrap pressed directly on the surface of the guacamole and refrigerate—preferably for no more than a few hours.",
+    "When you're ready to serve, scoop the guacamole into a serving bowl and garnish with the remaining diced mango and cilantro sprigs. Serve with tortilla chips, slices of cucumber or jicama."
+  ],
+  "yields": "2 1/2 cups",
```
Comment (jknndy) [inline tests/test_data/rickbayless.com/rickbayless.json]:
```suggestion
  "yields": "2 cups",
```

File: tests/test_data/rickbayless.com/rickbayless.json [hunk 6]
```diff
@@ -0,0 +1,24 @@
...
+  "description": "Recipe from Season 6, Mexico—One Plate at a Time",
+  "image": "https://www.rickbayless.com/wp-content/uploads/2013/12/Screen-Shot-2014-04-01-at-12.12.04-PM.png"
```
Comment (jknndy) [inline tests/test_data/rickbayless.com/rickbayless.json:25]:
```suggestion
  "total_time": null,
  "image": "https://www.rickbayless.com/wp-content/uploads/2013/12/Screen-Shot-2014-04-01-at-12.12.04-PM.png"
```

**Review by jknndy — DISMISSED** [review]
Hi @joeygerovac , thanks for the contribution! I made some comments throughout for your review.
